### PR TITLE
[DS-3967] 5x - Migrate Dockerfile to DSpace/DSpace

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3967

Derived from https://github.com/DSpace/DSpace/pull/2134. Inserts a build.properties file instead of a local.cfg file.

### CODE REVIEW

File: build.properties.docker [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# DSpace build.properties
```
Comment (tdonohue) [inline build.properties.docker]: As noted in today's meeting, let's move this to `[src]/src/main/docker/build.properties`

**Comment by ppmdo** [comment]
Tested on Fedora Linux 27. Build and compose work without problems.

Used this docker-compose file: https://github.com/DSpace-Labs/DSpace-Docker-Images/blob/master/docker-compose-files/dspace-compose/docker-compose.yml

**Review by pnbecker — APPROVED** [review]
:+1:
Reviewed, not tested.

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me. Merging

**Pull request merged**